src/codegen/llvm.zig

```diff
@@ -568,6 +580,40 @@ pub const LLVMIRModule = struct {
         return null;
     }
 
+    /// TODO: I expect both 'genIsNull' and 'genIsNonNull' to receive the same types,
+    /// however 'genIsNull' takes an optional value and 'genIsNonNull' takes a
+    /// pointer to an optional value.
+    fn genIsNull(self: *LLVMIRModule, inst: *Inst.UnOp) !?*const llvm.Value {
+        const operand = try self.resolveInst(inst.operand);
+        return self.builder.buildNot(self.builder.buildExtractValue(operand, 1, ""), "");
+    }
+
+    fn genIsNonNull(self: *LLVMIRModule, inst: *Inst.UnOp) !?*const llvm.Value {
+        const operand = try self.resolveInst(inst.operand);
+
+        const index_type = self.context.intType(32);
+
+        var indices: [2]*const llvm.Value = .{
+            index_type.constNull(),
+            index_type.constInt(1, false),
+        };
+
+        return self.builder.buildLoad(self.builder.buildInBoundsGEP(operand, &indices, 2, ""), "");
+    }
```

Comment:
@Vexu why do these 2 instructions receive different types of operands? (optional vs pointer to optional)

Reply:
I just copied what stage1 does. There are some TODOs regarding this in `astgen.zig`.